src/components/AddItemForm.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { format } from 'prettier';
+import { withFirestore } from 'react-firestore';
+import '../CSS/AddItemForm.css';
+class AddItemForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      next_purchase: null,
```

Comment:
I couldn't find a record of it the ticket or AC, but is there supposed to be a default value for `next_purchase`?

Reply:
The `next_purchase` state is not updating and saving properly to the firestore database. Your item name is updating properly. To have the select drop-down update the proper numerical value to the database, there are a couple of things you need to do.

```
this.handleSchdeule = this.handleSchdeule.bind(this);

  handleSchdeule(event) {
    this.setState({ 
      next_purchase: event.target.value,
     });
  }
// in the form
          <select value={this.state.value}

            onChange={this.handleSchdeule}> // add this like you have on the text field
            <option value="7">Soon</option>
            <option value="14">Kind Of Soon</option>
            <option value="30">Not Soon</option>
          </select>
```

Reply:
The `select` element expects to have _some_ value passed to it. `null` throws an error: 

![](https://cdn.zappy.app/09c7eecd69cbd3614f1f0a15fbc94af3.png)

I would make the default `14` (kinda soon).